index.js

```diff
@@ -13,19 +13,22 @@ const isAvailable = options => new Promise((resolve, reject) => {
 	});
 });
 
-const getPort = (options = {}) => new Promise((resolve, reject) => {
+const getPort = (options = {}) => {
 	if (typeof options.port === 'number') {
 		options.port = [options.port];
 	}
 
-	options.port.reduce((seq, port) => {
-		return seq.catch(() => {
-			return isAvailable(Object.assign({}, options, {port}))
-				.then(port => port)
-				.catch(Promise.reject.bind(Promise));
-		});
-	}, Promise.reject()).then(resolve).catch(reject);
-});
+	if (!options.port) {
+		options.port = [];
```

Comment:
Here you are modifying a user-supplied object, which could cause weird bugs on subsequent calls. Instead just do `return (options.port || []).reduce` on line 25.

Reply:
Yes

Reply:
Should I fix it in this pr or another one?

Reply:
I can fix it after merging this.